src/main/java/uk/org/tombolo/importer/phe/AdultObesityImporter.java

```diff
@@ -61,65 +57,165 @@ public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Except
-        RowCellExtractor subjectExtractor = new RowCellExtractor(1, CellType.STRING);
-        ConstantExtractor timestampExtractor = new ConstantExtractor(year);
 
-        SubjectType subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
+        // We create SubjectType object that we will use to get the appropriate geometries
+        // from OaImporter class
+        SubjectType localauthority = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
                 OaImporter.OaType.localAuthority.name(), OaImporter.OaType.localAuthority.datasourceSpec.getDescription());
-        for (AttributeLabel attributeLabel : AttributeLabel.values()){
-            ConstantExtractor attributeExtractor = new ConstantExtractor(attributeLabel.name());
-            RowCellExtractor valueExtractor
-                    = new RowCellExtractor(getAttributeColumnId(attributeLabel), CellType.NUMERIC);
-            timedValueExtractors.add(new TimedValueExtractor(
-                    getProvider(),
-                    subjectType,
-                    subjectExtractor,
-                    attributeExtractor,
-                    timestampExtractor,
-                    valueExtractor));
+
+        // We create an empty list that will keep our .csv values
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        CSVFormat format = CSVFormat.DEFAULT;
+        String fileLocation = getDatasourceSpec("adultObesity").getUrl();
+
+        // The code below fetches the .csv file from the URL we specified in our DatasourceSpec object
+        URL url;
+        try {
+            url = new URL(fileLocation);
+        } catch (MalformedURLException e) {
+            File file;
+            if (!(file = new File(fileLocation)).exists()) {
+                log.error("ERROR: File does not exist: " + fileLocation);
+            }
+            url = file.toURI().toURL();
+        }
+
+        InputStreamReader isr = new InputStreamReader(
+                downloadUtils.fetchInputStream(url, getProvider().getLabel(), ".csv"));
+
+        // Parsing our csv file
+        CSVParser csvFileParser = new CSVParser(isr, format);
+        csvRecords = csvFileParser.getRecords();
+
+
+        // We discard the first 6 records in our data file as these don't hold any meaningfull information.
+        // We do this  calling an iterator object and simply ignoring them:
+        Iterator<CSVRecord> rowIterator = csvRecords.iterator();
+
+        // skipping first 6 rows
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+        rowIterator.next();
+
+        // Looping through the rows of the .csv file
+        while (rowIterator.hasNext()){
+            CSVRecord row = rowIterator.next();
+            try{
+                Subject subject = SubjectUtils.getSubjectByTypeAndNameUnique(localauthority, String.valueOf(row.get(2)).trim());
```

Comment:
This will throw an `NonUniqueResultException` if there is more than one result.

Reply:
Same here...